src/arch/isa_parser/isa_parser.py

```diff
@@ -1078,6 +1078,16 @@ def split(sec):
         # next split's #define from the parser and add it to the current
         # emission-in-progress.
         try:
+            # x86 requires more python files to be interpreted. To allow this
+            # script to be run from other locations for x86, force adding the
+            # x86 ISA directory to the Python path.
+            import os
+            import sys
+
+            current_file_path = os.path.realpath(__file__)
+            arch_dir = os.path.dirname(os.path.dirname(current_file_path))
+            sys.path.append(arch_dir)
+            sys.path.append(os.path.join(arch_dir, "x86/isa/"))
```

Comment:
Hi Jason, I don't particularly understand how split is used in the ISA parser, but I can see it being referred by Arm ISA files as well...

Reply:
I don't understand the question.

The code highlighted is x86-specific. The reason it's needed is the `exec(split_setup + fixPythonIndentation(t[2]), self.exportContext)` executes this code: `from insts import microcode` (https://github.com/gem5/gem5/blob/stable/src/arch/x86/isa/microasm.isa#L53).

So, to `exec` the `let` block in microasm.isa, we need to ensure that the pythonpath includes the `isa` directory.

*I do not like this code, and I think that all cases of `exec` should be purged from the codebase.* However, I tried a couple of other quick fixes, but they didn't work.

I see no reason the code highlighted won't work for Arm because I have tested building x86, Arm, and RISC-V outside of scons with this code in a fully hermetic environment (e.g., bazel).